### DISCUSSION TITLE
Upgrade from version 5.0.24 to version 8.1.2

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#7102] [Help]
Please I have a working Serenity application written in version 5.0.24, is it possible to upgrade to the latest version of Serenity, or i have to rewrite the whole application using the latest template.
If its possible, is there a single document that shows the upgrade steps that I can follow.

![image](https://github.com/serenity-is/Serenity/assets/50826372/a4c1fbfd-a4f1-4b61-a66f-d7fe84eb412c)

Thank you

**Comment by edwardch** [comment]
Check out the migration guide docs, starting at this point - https://serenity.is/docs/migration/v5-0-20-to-v5-0-26

Then from 6.x onwards, check changelog, or there may be some 6-8 migration docs in the works (check the docs repo as well)

**Reply by rhemasystems** [reply]
Thank you @edwardch, I have been able to follow through to upgrade to version 6.9.6.
However I get the below issue for all grids in my project and am sure i may have missed out a step in the upgrade process.
Reduced grid height on all pages as well as distorted `Quick Search` control,distorted `Apply Changes` button
Please any help will be much appreciated

![image](https://github.com/serenity-is/Serenity/assets/50826372/67535c95-9700-414a-93a5-5b1a16de3c8f)

![image](https://github.com/serenity-is/Serenity/assets/50826372/08f552c9-2e8d-45fe-8bb0-db0faabfac52)

**Reply by dfaruque** [reply]
To fix grid height in xyzPage.ts, pass setHeight option.
` initFullHeightGridPage(new xyzGrid($('#GridDiv')).element, { setHeight: true });`

**Reply by rhemasystems** [reply]
Thank you @dfaruque , does it mean i have to do this across all the xyxPage.ts files in my project?

**Reply by dfaruque** [reply]
Yes, but it should be the default behavior.

**Comment by rhemasystems** [comment]
Hi @dfaruque, thank you, it worked ,but I have to do it for all xyzPage.ts, its not the default  behaviour for my project.
Isnt there a way to fix that?

Thank you
